### PR TITLE
Initial work on option lenght selection

### PR DESCRIPTION
**Pull request by Managor**
Initial ideas on how to implement the shortform/longform print modes.

### CODE REVIEW

**Comment by Managor** [comment]
Related issue https://github.com/tldr-pages/tldr/issues/13556

**Comment by sbrl** [comment]
Seems interesting. Does this always hold up with all instances of `{{a|b}}` where `{{-a|--b}}`?

What would you envision the CLI to look like for `tldr`? Could you disable the short/longform parsing support and display everything at all? What would the default be?

File: tldr.py [hunk 1]
```diff
@@ -478,6 +478,13 @@ def emphasise_example(x: str) -> str:
             line = line.replace(r'\{\{', '__ESCAPED_OPEN__')
             line = line.replace(r'\}\}', '__ESCAPED_CLOSE__')
 
+            # Extract long or short options from placeholders
+            if not (shortform and longform):
```
Comment (Managor) [inline tldr.py:482]: Default would obiously be True for both so that this condition causes a skip. I'm not certain what you're asking in the rest of the comment.

**Comment by Managor** [comment]
If a tool doesn't follow this convention then it won't be affected by this. Simple as that.

**Comment by acuteenvy** [comment]
This regular expression won't work if there are multiple short options in a single placeholder.
Example:
https://github.com/tldr-pages/tldr/blob/3823ba7fc64e2c368e87950f1aaf407d88f6957f/pages/common/rsync.md#L17
```md
`rsync {{-zvhP|--compress --verbose --human-readable --partial --progress}} {{path/to/source}} {{path/to/destination}}`
```

**Comment by Managor** [comment]
That is by design. Like I said in the original post, the initial implementation should be as strict as possible.

**Comment by Managor** [comment]
Alright, I'm looking for feedback again. Is this the proper way to implement options? Also how can I implement TLDR_SHORTFORM and TLDR_LONGFORM env variables?

**Comment by Managor** [comment]
Nvm figured it out. Now I'm just looking for feedback.

**Comment by Managor** [comment]
I adjusted the client specs in https://github.com/tldr-pages/tldr/pull/15253 and adjusted the code to match it. I'm looking for feedback.

**Comment by Managor** [comment]
Demonstrative images
Input:
image

tldr rsync
image

tldr -S rsync
image

tldr -V rsync
image

tldr -VS rsync
image

**Comment by Managor** [comment]
`grep -rE "{{\[.*\|.*\]}}"` shows no matches. Accidental usages of this pattern should not be found.

**Comment by Managor** [comment]
`grep -rE "{{-.*\|--.*}}"` to find all the current implementations.

**Comment by Managor** [comment]
@kbdharun You seem to be fluent in manpage language. Would you be willing to make an addition to the tldr manpage after this goes through?

**Comment by kbdharun** [comment]
> @kbdharun You seem to be fluent in manpage language. Would you be willing to make an addition to the tldr manpage after this goes through?

Sure

**Comment by Managor** [comment]
Updated `tldr -VS rsync`
image

**Comment by Managor** [comment]
I implemented the final command line options seen in the release. I'm looking for feedback again.

File: tldr.py [hunk 2]
```diff
@@ -675,7 +705,7 @@ def main() -> None:
                     " send a pull request to: https://github.com/tldr-pages/tldr"
                 ).format(cmd=command))
             else:
-                output(results[0][0], plain=options.markdown)
+                output(results[0][0], short, long, plain=options.markdown)
```
Comment (Managor) [inline tldr.py:708]: To me this doesn't feel like a proper way to pass the booleans.

File: README.md [hunk 3]
```diff
@@ -86,6 +86,8 @@ export TLDR_CACHE_ENABLED=1
 export TLDR_CACHE_MAX_AGE=720
 export TLDR_PAGES_SOURCE_LOCATION="https://raw.githubusercontent.com/tldr-pages/tldr/main/pages"
 export TLDR_DOWNLOAD_CACHE_LOCATION="https://tldr-pages.github.io/assets/tldr.zip"
+export TLDR_SHORT_OPTIONS=1
+export TLDR_LONG_OPTIONS=1
```
Comment (acuteenvy) [inline README.md:90]: How about changing this to just one env var: `TLDR_OPTIONS=short/long/both`?
https://github.com/tldr-pages/tldr/pull/15253#discussion_r1957144978

Comment (Managor) [reply README.md:90]: Thanks, I completely forgot that that discussion.

File: tldr.py [hunk 4]
```diff
@@ -623,7 +640,22 @@ def main() -> None:
     parser = create_parser()
 
     options = parser.parse_args()
-
+    short = False
+    long = True
```
[the remark this answers is not in the document]
Comment (Managor) [reply tldr.py:644]: What part of the readme? The example configuration? It's marked as `short` there because the purpose of a configuration is to alter the default behavior.

File: README.md [hunk 5]
```diff
@@ -180,3 +181,7 @@ can either use the `--source` flag when using tldr or by specifying the followin
   - it can also point to a local directory using `file:///path/to/directory`.
 - `TLDR_DOWNLOAD_CACHE_LOCATION` to control where to pull a zip of all pages from.
   - defaults to `https://github.com/tldr-pages/tldr/releases/latest/download/tldr.zip`.
+
+### Command options
+
+Pages might contain `{{[*|*]}}` pattern to let the client decice whether to show shortfrom or longform versions of options. This can be configured with `TLDR_OPTIONS` and it accepts values `short`, `long` and `both`
```
Comment (acuteenvy) [inline README.md:187]:
```suggestion
Pages might contain `{{[*|*]}}` patterns to let the client decide whether to show shortform or longform versions of options. This can be configured with `TLDR_OPTIONS`, which accepts values `short`, `long` and `both`
```

**Comment by Managor** [comment]
@sebastiaanspeck I would like to write tests for option placeholders. Do you have any tips the process and where to start?

**Review by kbdharun — APPROVED** [review]
LGTM, thanks for the additions. Tested the changes locally and it works as expected.
